services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java

```diff
@@ -483,6 +484,87 @@ public Response updateNews(@Context HttpServletRequest request,
     }
   }
 
+  @PUT
+  @Path("updateNewsDraftVisibility/{newsId}")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @ApiOperation(value = "Update the current draft visibility", httpMethod = "PUT", response = Response.class, notes = "This updates the news if the authenticated user is a member of the space or a spaces super manager.")
+  @ApiResponses(value = { @ApiResponse(code = 200, message = "News updated"),
+      @ApiResponse(code = 400, message = "Invalid query input"),
+      @ApiResponse(code = 401, message = "User not authorized to update the darft visibility"),
+      @ApiResponse(code = 500, message = "Internal server error") })
+  public Response updateNewsDraftVisibility(@Context HttpServletRequest request,
+                             @ApiParam(value = "news id", required = true) @PathParam("newsId") String newsId,
+                             @ApiParam(value = "draft visibility id", required = true) String draftVisibility) {
+
+    if (newsId == null) {
+      return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+    if (draftVisibility == null) {
+      return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+    try {
+      newsService.setNewsDraftVisibility(newsId, draftVisibility);
```

Comment:
security : first thing to do before any business logic is to check for permissions for user.  HTTP 401 is never raised,

Reply:
the PR is always under dev and not yet validated